src/util-logopenfile.c

```diff
@@ -73,6 +76,49 @@ SCLogOpenUnixSocketFp(const char *path, int sock_type)
     return NULL;
 }
 
+/*
+ * Attempt to reconnect a disconnected (or never-connected) Unix domain socket.
+ *  \retval 1 if it is now connected; otherwise 0
+ */
+static int SCLogUnixSocketReconnect(LogFileCtx *log_ctx)
+{
+    int disconnected = 0;
+    if (log_ctx->fp) {
+        SCLogError(SC_ERR_SOCKET,
+            "Write error on Unix socket \"%s\": %s; reconnecting...",
+            log_ctx->filename, strerror(errno));
+        fclose(log_ctx->fp);
+        log_ctx->fp = NULL;
+        log_ctx->reconn_timer = 0;
+        disconnected = 1;
+    }
+
+#if defined(_POSIX_TIMERS) && _POSIX_TIMERS > 0 && _POSIX_MONOTONIC_CLOCK
```

Comment:
On 07/16/2015 01:57 PM, jsbarber wrote:
> I tend to avoid gettimeofday due to time adjustment hassles (time going
> backward and so forth). I suppose that issue could be ignored here.
> 

I think so ya, we use it in other places as well :)